File: tlv/src/test/java/dev/keiji/tlv/BerTlvEncoderTest.java
package dev.keiji.tlv;

import static org.junit.Assert.assertArrayEquals;
import static org.junit.Assert.fail;

import org.junit.Test;

public class BerTlvEncoderTest {
    @Test
    public void convertToLengthTest1() {
        byte[] expected = new byte[]{126};
        byte[] actual = BerTlvEncoder.convertToLength(126);
        assertArrayEquals(expected, actual);
    }

    @Test
    public void convertToLengthTest2() {
        byte[] expected = new byte[] {(byte) 0b10000001, (byte) 128};
        byte[] actual = BerTlvEncoder.convertToLength(128);
        assertArrayEquals(expected, actual);
    }

    @Test
    public void convertToLengthTest_Exception1() {
        try {
            byte[] actual = BerTlvEncoder.convertToLength(-1);
            fail();
        } catch (IllegalArgumentException exception) {
            System.out.println(exception);
        }
    }
}

File: tlv-ksp/src/test/java/dev/keiji/tlv/BerTlvEncoderTest.kt
package dev.keiji.tlv

import org.junit.Assert
import org.junit.Test
import java.lang.IllegalArgumentException

class BerTlvEncoderTest {
    @Test
    fun convertToLengthTest1() {
        val expected = byteArrayOf(126)
        val actual = BerTlvEncoder.convertToLength(126)
        Assert.assertArrayEquals(expected, actual)
    }

    @Test
    fun convertToLengthTest2() {
        val expected = byteArrayOf(0b10000001.toByte(), 128.toByte())
        val actual = BerTlvEncoder.convertToLength(128)
        Assert.assertArrayEquals(expected, actual)
    }

    @Test
    fun convertToLengthTest_Exception1() {
        try {
            val actual = BerTlvEncoder.convertToLength(-1)
            Assert.fail()
        } catch (exception: IllegalArgumentException) {
            println(exception)
        }
    }
}
